lib/supabase.ex

```diff
@@ -166,8 +166,37 @@ defmodule Supabase do
     apply(__MODULE__, which, [])
   end
 
-  @json_library Application.compile_env(:supabase, :json_library, Jason)
+  @json_library Application.compile_env(:supabase_potion, :json_library, Jason)
 
   @doc "Returns the configured JSON encoding library for Supabase libraries."
+  @spec json_library :: Poison | Jason | JSON
   def json_library, do: @json_library
+
+  @doc false
+  defguardp is_atom_opt(atom) when atom in ~w(atoms atoms!)a
+
+  @doc false
+  def decode_json(term, opts) do
```

Comment:
will be used internally on the other libraries to avoid duplicating this logic across the ecosystem